arches_lingo/src/arches_lingo/components/generic/LabelEditor.vue

```diff
@@ -0,0 +1,390 @@
+<script setup lang="ts">
+import { computed, inject, onMounted, ref, toRaw, toRef, type Ref } from "vue";
+
+import Button from "primevue/button";
+import { useGettext } from "vue3-gettext";
+import { useRoute, useRouter } from "vue-router";
+import { useToast } from "primevue/usetoast";
+
+import { fetchLists } from "@/arches_references/api.ts";
+
+import {
+    createScheme,
+    createSchemeLabel,
+    fetchGroupRdmSystemList,
+    fetchPersonRdmSystemList,
+    fetchTextualWorkRdmSystemList,
+    updateSchemeLabel,
+} from "@/arches_lingo/api.ts";
+
+import DateDatatype from "@/arches_lingo/components/generic/DateDatatype.vue";
+import NonLocalizedString from "@/arches_lingo/components/generic/NonLocalizedString.vue";
+import ReferenceDatatype from "@/arches_lingo/components/generic/ReferenceDatatype.vue";
+import ResourceInstanceRelationships from "@/arches_lingo/components/generic/ResourceInstanceRelationships.vue";
+
+import {
+    EDIT,
+    ERROR,
+    NEW,
+    selectedLanguageKey,
+} from "@/arches_lingo/constants.ts";
+
+import type {
+    AppellativeStatus,
+    ControlledListResult,
+    ControlledListItem,
+    ControlledListItemResult,
+    ResourceInstanceReference,
+    ResourceInstanceResult,
+    SchemeInstance,
+} from "@/arches_lingo/types.ts";
+import type { Language } from "@/arches_vue_utils/types.ts";
+
+const emit = defineEmits(["update"]);
+const toast = useToast();
+const route = useRoute();
+const router = useRouter();
+const selectedLanguage = inject(selectedLanguageKey) as Ref<Language>;
+const { $gettext } = useGettext();
+const formId = (route.params.id as string) ?? Math.random().toString();
```

Comment:
TIL that Vue 3.5 ships a `useId()` hook for this: https://vuejs.org/api/composition-api-helpers.html#useid

That's likely better than rolling our own.

Also: why wouldn't we put this label inside the `<DateDatatypeEditor>` and perhaps use `<label for=` like you had before, instead of passing this down through all these layers?

Reply:
Pushed an update to use this